experimental/Schemes/src/Resolution_structure.jl

```diff
@@ -125,12 +130,25 @@ end
 ##################################################################################################
 underlying_morphism(phi::NormalizationMorphism) = phi.underlying_morphism
 inclusion_morphisms(phi::NormalizationMorphism) = phi.inclusions
-normalization_steps(phi::NormalizationMorphism) = phi.normalization_steps
 morphisms(phi::AbsDesingMor) = copy(phi.maps)
 morphism(phi::AbsDesingMor,i::Int) = phi.maps[i]
 last_map(phi::AbsDesingMor) = phi.maps[end]
+normalization_steps(phi::MixedBlowUpSequence) = phi.normalization_steps
 
 exceptional_divisor_list(phi::BlowUpSequence) = phi.ex_div
+exceptional_divisor_as_ideal_sheafs(phi::MixedBlowUpSequence) = exceptional_divisor_list(phi,true)
+
+function exceptional_divisor_as_ideal_sheafs(phi::BlowUpSequence)
+  !is_empty(phi.ex_div) || return phi.ex_div
+  E_list = phi.ex_div
+  ret_list = Vector{AbsIdealSheaf}()
+  for i in 1:length(E_list)
+    E = E_list[i]
+    E = (!(E isa AbsIdealSheaf) ? ideal_sheaf(E) : E)
```

Comment:
Maybe avoid the negation?
```suggestion
    E = E isa AbsIdealSheaf ? E : ideal_sheaf(E)
```
```suggestion
    E = (!(E isa AbsIdealSheaf) ? ideal_sheaf(E) : E)
```

Reply:
No, I prefer writing this the way it would be stated (If it isn't an  ideal_sheaf, force it to be one)